hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDiffUtils.java

```diff
@@ -50,5 +62,48 @@ public static String constructBucketKey(String keyName) {
     return builder.toString();
   }
 
+  public static void filterRelevantSstFiles(Set<String> inputFiles,
+      Map<String, String> tableToPrefixMap) {
+    for (Iterator<String> fileIterator =
+         inputFiles.iterator(); fileIterator.hasNext();) {
+      String filepath = fileIterator.next();
+      if (!RocksDiffUtils.doesSstFileContainKeyRange(filepath,
+          tableToPrefixMap)) {
+        fileIterator.remove();
+      }
+    }
+  }
+
+  public static boolean doesSstFileContainKeyRange(String filepath,
+      Map<String, String> tableToPrefixMap) {
+    try (SstFileReader sstFileReader = new SstFileReader(new Options())) {
+      sstFileReader.open(filepath);
+      TableProperties properties = sstFileReader.getTableProperties();
+      String tableName = new String(properties.getColumnFamilyName(), UTF_8);
+      if (tableToPrefixMap.containsKey(tableName)) {
+        String prefix = tableToPrefixMap.get(tableName);
+        SstFileReaderIterator iterator =
+            sstFileReader.newIterator(new ReadOptions());
+        iterator.seekToFirst();
+        String firstKey = RocksDiffUtils.constructBucketKey(
+            new String(iterator.key(), UTF_8));
+        iterator.seekToLast();
+        String lastKey = RocksDiffUtils.constructBucketKey(
+            new String(iterator.key(), UTF_8));
+        boolean keyWithPrefixPresent =
+            RocksDiffUtils.isKeyWithPrefixPresent(prefix, firstKey, lastKey);
+        if (!keyWithPrefixPresent) {
```

Comment:
why can't we just return RocksDiffUtils.isKeyWithPrefixPresent(prefix, firstKey, lastKey);

Reply:
Done.